composable.js

```diff
@@ -287,29 +287,52 @@
     Composable.prototype.getTransformation = function (transformationInput) {
         if (isFunction(transformationInput)) {
             return transformationInput;
-        } else if (!isString(transformationInput)) {
+        } else if (!isArray(transformationInput) && !isString(transformationInput)) {
             throw 'Invalid Input Type: ' + typeof transformationInput;
         }
 
-        if (!transformationInput) {
+        if (!transformationInput || !transformationInput.length) {
             // either command string is empty or no sub command given in recursion
             return null;
-        } else if (this.T.hasOwnProperty(transformationInput)) {
-            return this.T[transformationInput];
         }
 
-        var transformationArray = transformationInput.split(':');
-        var command = transformationArray.shift();
-        var commandArgString = transformationArray.join(':');
+        var command, subCommand, args;
+        if (isArray(transformationInput)) {
+            if (transformationInput.length === 1 && this.T.hasOwnProperty(transformationInput[0])) {
+                return this.T[transformationInput[0]];
+            }
 
-        var subCommand;
-        if (arrayTransformations.hasOwnProperty(command)) {
-            subCommand = this.getTransformation(commandArgString);
-        } else if (!this.T.hasOwnProperty(command)) {
-            return null;
+            command = transformationInput[0];
+            if (!this.T.hasOwnProperty(command)) {
+                return null;
+            }
+
+            args = transformationInput.slice(1);
+            if (arrayTransformations.hasOwnProperty(command)) {
+                subCommand = this.getTransformation(args);
+                if (subCommand) {
+                    args = [subCommand];
+                }
+            }
+        } else {
```

Comment:
Maybe split this into a private method, but it doesn't seem necessary unless more additions to this are planned.